src/Swarrot/Processor/XDeath/XDeathMaxLifetimeProcessor.php

```diff
@@ -117,8 +117,7 @@ private function handleException($exception, Message $message, array $options)
                 );
             } elseif (isset($queueXDeathHeader['time'])) {
                 $xDeathTimestamp = $queueXDeathHeader['time'];
-                // PhpAmqpLib compatibility
-                if ($xDeathTimestamp instanceof \DateTime) {
+                if (method_exists($xDeathTimestamp, 'getTimestamp')) {
```

Comment:
I think having `$xDeathTimestamp instanceof \DateTime || $xDeathTimestamp instanceof \AMQPTimestamp` might be better to see what is actually supported